dandiapi/api/tests/test_asset.py

```diff
@@ -357,6 +357,65 @@ def test_asset_rest_update(api_client, user, draft_version, asset, asset_blob):
     assert start_time < end_time
 
 
+@pytest.mark.django_db
+def test_asset_rest_update_only_metadata(api_client, user, draft_version, asset):
+    assign_perm('owner', user, draft_version.dandiset)
+    api_client.force_authenticate(user=user)
+    draft_version.assets.add(asset)
+
+    new_path = 'test/asset/rest/update.txt'
+    new_metadata = {
+        'schemaVersion': '0.4.4',
+        'encodingFormat': 'application/x-nwb',
+        'path': new_path,
+        'foo': 'bar',
+        'num': 123,
+        'list': ['a', 'b', 'c'],
+    }
+
+    resp = api_client.put(
+        f'/api/dandisets/{draft_version.dandiset.identifier}/'
+        f'versions/{draft_version.version}/assets/{asset.asset_id}/',
+        {'metadata': new_metadata},
+        format='json',
+    ).data
+    new_asset = Asset.objects.get(asset_id=resp['asset_id'])
+    assert resp == {
+        'asset_id': UUID_RE,
```

Comment:
is there some magic which I am not aware here -- `==` would do `re.match`if it detects a value be a regex instance ???

Reply:
Yep, it's magic. To make asserting large JSON blobs easier, we implemented [this class](https://github.com/dandi/dandi-api/blob/master/dandiapi/api/tests/fuzzy.py) which evaluates as being equal to any string that matches the initialized regex. `UUID_RE` specifically is equal to any UUID string.

Reply:
interesting! thanks for the pointer!  attn @jwodder -- may be he would like to adopt this too

Reply:
I just recently wrote [a library of things like this](https://github.com/jwodder/anys) after people on IRC said it hadn't been done.

Reply:
awesome -- "Great minds think alike" ;)